data-imports/clean_utils.py

```diff
@@ -51,13 +51,44 @@ def clean_boro(table, column, replacements):
     """
     return '''
     UPDATE {table} SET {column} = preg_replace('/{mn}/i', 'MN', {column});
+    UPDATE {table} SET {column} = preg_replace('/{bx}/i', 'BX', {column});
     UPDATE {table} SET {column} = preg_replace('/{bk}/i', 'BK', {column});
-    UPDATE {table} SET {column} = preg_replace('/{si}/i', 'SI', {column});
     UPDATE {table} SET {column} = preg_replace('/{qn}/i', 'QN', {column});
-    UPDATE {table} SET {column} = preg_replace('/{bx}/i', 'BX', {column});
+    UPDATE {table} SET {column} = preg_replace('/{si}/i', 'SI', {column});
     '''.format(table=table, column=column, mn=replacements["mn"], bk=replacements["bk"],
             si=replacements["si"], qn=replacements["qn"], bx=replacements["bx"])
 
+def add_boroid(table, column):
+    """ Returns a SQL statement that adds a column for boro IDs (used in BBL) and
+        populates via conversion from a cleaned boro column.
+    """
+    columnid = column + 'id'
+    sql = ''
+    lookup_table = 'boroid_lookup'
+    if not table_exists(lookup_table):
+        sql = sql + """
+        CREATE TABLE {table} (
+            boro_code varchar(2),
+            boro_id int,
+            PRIMARY KEY(boro_code)
```

Comment:
Not really - we're already running all the boro columns through the "clean_boro" SQL above, which replaces everything with the two-letter codes.

This table is also only meant to be used for this particular step, so it could probably be a temporary table, though it's not really taking much space as is.